crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -1725,10 +1703,10 @@ impl<'a> Iter<'a> {
 enum ScanStage<'a> {
     Start,
     CurrentTx {
-        iter: std::collections::btree_map::Iter<'a, RowId, ProductValue>,
+        iter: indexmap::map::Iter<'a, RowId, ProductValue>,
     },
     Committed {
-        iter: std::collections::btree_map::Iter<'a, RowId, ProductValue>,
+        iter: indexmap::map::Iter<'a, RowId, ProductValue>,
```

Comment:
Yeah I'm not sure either; I actually went with plain HashMap first but I suspect we do want some determinism since this iteration is exposed to users, and we tend to try and keep all such APIs deterministic even if specific order is not guaranteed, so I switched to IndexMap.

The difference between these two is almost non-existant though, and we already depend on IndexMap in other places so it seems like reasonable tradeoff.

Reply:
I'd also err on keeping deterministic order for now. We should eventually be so precise in our documentation about what such guarantees we provide :)